tex/generic/pgf/utilities/pgfcalendar.code.tex

```diff
@@ -797,5 +802,207 @@
+  leap year/.code={%
+    \pgfutil@tempcnta=#1\relax
+    \divide\pgfutil@tempcnta4
+    \multiply\pgfutil@tempcnta4
+    \ifnum\pgfutil@tempcnta=#1\relax
+      \divide\pgfutil@tempcnta100
+      \multiply\pgfutil@tempcnta100
+      \ifnum\pgfutil@tempcnta=#1\relax
+        \divide\pgfutil@tempcnta400
+        \multiply\pgfutil@tempcnta400
+        \ifnum\pgfutil@tempcnta=#1\relax
+          \pgfcalendarmatchestrue
+        \fi
+      \else
+        \pgfcalendarmatchestrue
+      \fi
+    \fi},
+  leap year/.default=\pgfcalendarifdateyear,
+  not/.code=% true when #1 == false
+    \begingroup
+      \pgfcalendar@launch@ifdate{#1}{\let\pgf@cal@temp\pgfutil@empty}{\def\pgf@cal@temp{\pgfcalendarmatchestrue}}%
+    \expandafter\endgroup\pgf@cal@temp,
+  and/.value required,
+  and/.code=% and = {<cond 1>, <cond 2>, <cond 3>, …}
+    \begingroup
+      \pgfcalendarmatchestrue
+      \pgfqkeys{/pgf/calendar/and}{#1}%
+      \ifpgfcalendarmatches % is it still true?
+        \expandafter\pgfutil@firstoftwo
+      \else
+        \expandafter\pgfutil@secondoftwo
+      \fi
+      {\def\pgf@cal@temp{\pgfcalendarmatchestrue}}%
+      {\let\pgf@cal@temp\pgfutil@empty}%
+    \expandafter\endgroup\pgf@cal@temp,
+  and/.unknown/.code=% only inside the group of and/.code
+    \ifpgfcalendarmatches
+      \expandafter\pgfutil@firstofone
+    \else
+      \expandafter\pgfutil@gobble
+    \fi
+    {%
+      \begingroup
+        \pgfcalendar@launch@ifdate{\pgfkeyscurrentname={#1}}%
+        {\let\pgf@cal@temp\pgfutil@empty}{\def\pgf@cal@temp{\pgfcalendarmatchesfalse}}
+      \expandafter\endgroup\pgf@cal@temp
+    }%
+}
+
+%
+% Weeks
+%
+% stored the Monday of week 1 of the year #1 as julian date number
+% and the weekday of the first of the year in a macro called
+% \pgfcalendar@week@<year> so that calculations are only done
+% once per year
+\def\pgfcalendar@week@setup#1{%
+  \pgfutil@IfUndefined{pgfcalendar@week@#1}{%
+    \begingroup
+      \pgfcalendardatetojulian{#1-01-01}\pgfutil@tempcnta
+      \pgfcalendarjuliantoweekday\pgfutil@tempcnta\pgfutil@tempcntb
+      %
+      % tempcnta holds the julian number for first day of the current year
+      % tempcntb holds the weekday for the first day of the current year
+      %
+      % set tempcnta to the Monday of the week with first day of current year
+      %
+      \advance\pgfutil@tempcnta by -\pgfutil@tempcntb
+      %
+      % if the first week starts at Fri, Sat or Sun, next week is the 1st week
+      %
+      \ifnum\pgfutil@tempcntb>3\relax
+        \advance\pgfutil@tempcnta by 7\relax
+      \fi
+      %
+      \edef\pgf@cal@temp{\def\expandafter\noexpand\csname pgfcalendar@week@#1\endcsname
+        {{\the\pgfutil@tempcnta}{\the\pgfutil@tempcntb}}}%
+      \expandafter\endgroup\pgf@cal@temp
+  }{}%
+}
+
+% Will store the week of the Julian date count #1 of year #2 in count #3
+\def\pgfcalendarjulianyeartoweek#1#2#3{\pgfcalendarjulianyeartoweek@#1{#2}#3\iftrue}
```

Comment:
There is no advantage in omitting those braces. It will only create parsing ambiguities.
```suggestion
\def\pgfcalendarjulianyeartoweek#1#2#3{\pgfcalendarjulianyeartoweek@{#1}{#2}{#3}{\iftrue}}
```